test/FormLabelSpec.tsx

```diff
@@ -0,0 +1,169 @@
+        <FormLabel data-testid="test-id" />
+      </FormGroup>,
+    );
+
+    const element = getByTestId('test-id');
+    element.hasAttribute('for');
+    element.getAttribute('for').should.equal('foo');
+  });
+
+  it('should render as a Col', () => {
+    const { getByTestId } = render(
+      <FormLabel column sm={4} data-testid="test-id">
+        Label
+      </FormLabel>,
+    );
+
+    const element = getByTestId('test-id');
+    element.classList.length.should.equal(3);
+    element.classList.contains('form-label').should.be.true;
+    element.classList.contains('col-form-label').should.be.true;
+    element.classList.contains('col-sm-4').should.be.true;
+  });
+
+  it('should use controlId for htmlFor when render as Col', () => {
+    const { getByTestId } = render(
+      <FormGroup controlId="foo">
+        <FormLabel column sm={4} data-testid="test-id" />
+      </FormGroup>,
+    );
+
+    const element = getByTestId('test-id');
+    element.classList.length.should.equal(3);
+    element.classList.contains('form-label').should.be.true;
+    element.classList.contains('col-form-label').should.be.true;
+    element.classList.contains('col-sm-4').should.be.true;
+    element.hasAttribute('for');
+    element.getAttribute('for').should.equal('foo');
+  });
+
+  it('should respect visuallyHidden', () => {
+    const { getByTestId } = render(
+      <FormLabel visuallyHidden data-testid="test-id">
+        Label
+      </FormLabel>,
+    );
+
+    const element = getByTestId('test-id');
+    element.classList.length.should.equal(2);
+    element.classList.contains('visually-hidden').should.be.true;
+  });
+
+  it('should prefer explicit htmlFor', () => {
+    shouldWarn('ignored');
+
+    const { getByTestId } = render(
+      <FormGroup controlId="foo">
+        <FormLabel htmlFor="bar" data-testid="test-id" />
+      </FormGroup>,
+    );
+
+    const element = getByTestId('test-id');
+    element.hasAttribute('for');
+    element.getAttribute('for').should.equal('bar');
+  });
+
+  it('should support ref forwarding', () => {
+    let input;
+    class Container extends React.Component {
+      render() {
+        return (
+          <FormGroup controlId="foo">
+            <FormLabel
+              ref={(ref) => {
+                input = ref;
+              }}
+            />
+          </FormGroup>
+        );
+      }
+    }
+
+    render(<Container />);
+    input.tagName.toUpperCase().should.to.equal('LABEL');
+  });
+
+  it('should support ref forwarding when rendered as a Col', () => {
+    let input;
+    class Container extends React.Component {
+      render() {
+        return (
+          <FormGroup controlId="foo">
+            <FormLabel
+              column
+              ref={(ref) => {
+                input = ref;
+              }}
+            />
+          </FormGroup>
+        );
+      }
+    }
+
+    render(<Container />);
+    input.tagName.toUpperCase().should.to.equal('LABEL');
```

Comment:
`toLowerCase` would be better to match the other tests.  Alternatively, I believe the tag should always be capitalized so you don't need to call `toUpperCase`

Reply:
Fixed in the commit https://github.com/react-bootstrap/react-bootstrap/pull/6224/commits/023f9b3e2cffbb507001f992d76bb9cfb4243310